flang: Fails to link to compiler-rt

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$> brew config
HOMEBREW_VERSION: 4.4.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 55f399c10dc7e6cc07497bdfccdecc96329e3b57
Last commit: 8 days ago
Branch: stable
Core tap JSON: 27 Jan 14:36 UTC
Core cask tap JSON: 27 Jan 14:36 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.7 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.7/bin/ruby
CPU: deca-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 13.6.7-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
Rosetta 2: false

$> brew doctor
Your system is ready to brew.
```

### Verification

- [x] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [x] I ran `brew update` and am still able to reproduce my issue.
- [x] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [x] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I was trying to compile & link a Fortran program w/ `brew`-installed flang.

### What happened (include all command output)?

My program failed to link because the `flang` driver does not find the appropriate `compiler-rt` runtime to link to.

A minimal reproducer is a simple program that performs complex division:
```
$> echo reproducer.f90
Program main
  Complex :: y
  y = y/2
End Program
```

Compiling & linking with `flang-new reproducer.f90` fails with an undefined symbol error (typically provided by compiler-rt):
```
$> flang-new reproducer.f90
Undefined symbols for architecture arm64:
  "___divsc3", referenced from:
      __QQmain in reproducer-2a571c.o
ld: symbol(s) not found for architecture arm64
flang-new: error: linker command failed with exit code 1 (use -v to see invocation)
```

You can see that this symbol is provided by `libclang_rt.osx.a`:
```
$> nm /opt/homebrew/Cellar/llvm/19.1.7/lib/clang/19/lib/darwin/libclang_rt.osx.a | grep divsc3
/opt/homebrew/Cellar/llvm/19.1.7/lib/clang/19/lib/darwin/libclang_rt.osx.a(divsc3.c.o):
0000000000000000 T ___divsc3
```

Explicitly providing this library on the link line resolves the issue:
```
$> flang-new reproducer.f90 /opt/homebrew/Cellar/llvm/19.1.7/lib/clang/19/lib/darwin/libclang_rt.osx.a
$> echo $?
0
```

### What did you expect to happen?

I'd expect that the `flang` is installed such that it can discover & link to `libclang-rt.osx.a`.

This is likely just because `flang` is installed away from the rest of `llvm`. The `flang` driver _expects_ to be installed in the same directory tree as `clang`. If this formula can be configured to do so, that would be the preferred solution!

When I build & install LLVM myself (w/ both `clang` + `flang` installed in the same tree), the `flang` driver automatically links to compiler-rt. I get the same correct behaviour with the pre-built packages provided by https://github.com/llvm/llvm-project/releases/tag/llvmorg-19.1.7. 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
$> brew install flang
$> echo "Program main; Complex :: y; y = y/2; End Program" > reproducer.f90
$> flang-new reproducer.f90
```

--- a/Formula/f/flang.rb
+++ b/Formula/f/flang.rb
@@ -5,6 +5,7 @@ class Flang < Formula
   sha256 "82401fea7b79d0078043f7598b835284d6650a75b93e64b6f761ea7b63097501"
   # The LLVM Project is under the Apache License v2.0 with LLVM Exceptions
   license "Apache-2.0" => { with: "LLVM-exception" }
+  revision 1
   head "https://github.com/llvm/llvm-project.git", branch: "main"
 
   livecheck do
@@ -64,7 +65,9 @@ class Flang < Formula
     # Our LLVM is built with exception-handling, which requires linkage with the C++ standard library.
     # TODO: Remove this if/when we've rebuilt LLVM with `LLVM_ENABLE_EH=OFF`.
     flang_cfg_file = if OS.mac?
-      ["-lc++", "-Wl,-lto_library,#{llvm.opt_lib}/libLTO.dylib"]
+      resource_dir = Utils.safe_popen_read(llvm.opt_bin/"clang", "-print-resource-dir").chomp
+      resource_dir.gsub!(llvm.prefix.realpath, llvm.opt_prefix)
+      ["-lc++", "-Wl,-lto_library,#{llvm.opt_lib}/libLTO.dylib", "-resource-dir=#{resource_dir}"]
     else
       ["-lstdc++"]
     end
@@ -161,6 +164,14 @@ class Flang < Formula
 
     sorted_testresult = testresult.split("\n").sort.join("\n")
     assert_equal expected_result.strip, sorted_testresult.strip
+
+    (testpath/"runtimes.f90").write <<~FORTRAN
+      Program main
+        Complex :: y
+        y = y/2
+      End Program
+    FORTRAN
+    system bin/flang_driver, "-v", "runtimes.f90"
 
     return if OS.linux?
 

--- a/Formula/f/flang.rb
+++ b/Formula/f/flang.rb
@@ -13,13 +13,12 @@ class Flang < Formula
   end
 
   bottle do
-    rebuild 1
-    sha256 cellar: :any,                 arm64_sequoia: "88cd183a3fd754f57ba8b4637ea0f7eebf9ca5f225fc5eb951377a705d3bf7d8"
-    sha256 cellar: :any,                 arm64_sonoma:  "9bf49edca702bfe6554da248bbdeb305d82c4068f75508852064ea13659f99ec"
-    sha256 cellar: :any,                 arm64_ventura: "d9b25602d8deb833264d99dd4c4e051cf3e2dd44bfc3d41a3a1411555d4c89f2"
-    sha256 cellar: :any,                 sonoma:        "8429abc9db2a89890efa451c0a5a110021e7404d86291615e58805a015b956ff"
-    sha256 cellar: :any,                 ventura:       "be9edd442a2109b037409ab71af5ad00860e08ffe40f04f31f403db712465a37"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:  "db66d721fd417bd732de221ad2d430da6f1195f949d310bb2c115ff4cec6b44d"
+    sha256 cellar: :any,                 arm64_sequoia: "5192eee40409a81f900297dd882616a98ccf51b9667b435c1543843160b231d0"
+    sha256 cellar: :any,                 arm64_sonoma:  "cf420f494db818c4fab571f8811ea9de41c883b6509224eb571f10717688d98b"
+    sha256 cellar: :any,                 arm64_ventura: "47f853c166f592d8a45db005468a2556cda8faa3711375388e603617e44a8b42"
+    sha256 cellar: :any,                 sonoma:        "b68ac9224e17ccd56b5eb7bb14770d52129ab3d46d61ad1603004e26d033ee60"
+    sha256 cellar: :any,                 ventura:       "f570df992d9d2dbbb3440285bf5ffe0e5504c4dc94393d402c1bfbf15d15b2b5"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:  "e231af3919949f50fda2ad2e2173bbc6de77f5f386496cbecc96a7aecb713018"
   end
 
   depends_on "cmake" => :build

